frontend/storage/utils/fetchGameData.ts

```diff
@@ -0,0 +1,75 @@
+import { GameSeries } from '../models/GameSeries';
+import { Game } from '../models/Game';
+import { ApiRequests } from './apiRequestManager';
+import { GameEnvironment } from '../models/GameEnvironment';
+import { Environment } from '../models/Environment';
+import { DefaultImage, Image } from '../models/Image';
+import { AbilityExample } from '../models/AbilityExample';
+import { Ability } from '../models/Ability';
+import { AbilityGroup } from '../models/AbilityGroup';
+import { AbilityCategory } from '../models/AbilityCategory';
+import { createGameAbilitiesListData } from './createGameAbilitiesListData';
+
+export async function fetchGameViewData(id: number) {
+	try {
+		const game: Game = await ApiRequests.get(`games/${id}`, {});
+		const series: GameSeries = await ApiRequests.get(`game-series/${game.seriesId}`, {});
+		return { game, series };
+	} catch (error) {
+		throw new Error(error);
+	}
+}
+export async function fetchGameBoxData(gameId: number, imageId: number | undefined) {
+	try {
+		const gameEnvironments: Array<GameEnvironment> = await ApiRequests.get(`game-environments`, { gameId: gameId });
+		let image: Image;
+		if (imageId) {
+			image = await ApiRequests.get(`images/${imageId}`, {});
+		} else {
+			image = DefaultImage;
+		}
+		const environments: Array<Environment> = new Array();
+		for (const gameEnvironment of gameEnvironments) {
+			environments.push(await ApiRequests.get('environments/' + gameEnvironment.environmentId, {}));
+		}
+		return { image, environments };
+	} catch (error) {
+		throw new Error(error);
+	}
+}
+export async function fetchGameAbilitiesListData(gameId: number) {
+	try {
+		const abilityExamples: Array<AbilityExample> = await ApiRequests.get(`ability-examples`, { gameId: gameId });
+		const idsToFetch: Set<number> = new Set();
```

Comment:
You can simplify this a bit:
```
function unique<T>(values: Array<T>): Array<T> {
    const set = new Set(values);

    return Array.from(set.values());
}

const idsToFetch = unique(abilityExamples.map(x => x.abilityId));
```

(You can pass the array of values to the constructor of Set and it will automatically populate itself)

Or you can go even further, because it sounds like a pattern you'll do a lot:

```
function getUniqueValues<T>(models: Array<T>, field: keyof T): Array<T[keyof T]> {
    const values = models.map(x => x[field]);
    const set = new Set(values);

    return Array.from(set.values());
}

const idsToFetch = getUniqueValues(abilityExamples, 'abilityId');
```

[See an example](https://www.typescriptlang.org/play/?ssl=10&ssc=2&pln=5&pc=1#code/JYOwLgpgTgZghgYwgAgCIFcEGtkG8BQyRyAjuolgFzIDOYUoA5gNz4C+++M6ICYwAexDJGEMAFUQwMhABqcADboINADwAVAHwAKALYCAJhAU1qAQShQ4ATw2aANMhjBjB6lgjWBMZOoCU5pY2GgDaHl4+6gC6mniExAhCdMgAborKNMgAvMj6RiYAdLpwAA7aAB7ZseUhzq5RfqzEyIkgyTRi2cggEADuyADKYtppSiqNnM1QYuhQwhZW1gUwUAK62h1gBaMZ2n4THPityQaYWKZoZyFRXSHxRLik5NjUAOQAjK-IbPb3eE8UN4AJi+Pz+jzIgOQrxB31+zQhzyo0IALKD4cREVDXmi4fgoqxjmBkDxpMoAIpIzI5UQSKQyeRjGjaU7YGiOABEkOwHIm+AA9PzkABhBTAbDIDkAJR4HOQcBABhaAAsIBKvLMWkkBAoIEdtbqCgoBIxtKSZJSKDQ+YLkAB1AToBRKkACYmJXQlYC6ygCoVEkn05TC5XAKAlakiMSSMlydIqFlndmShCh8O85hAA)